src/ast/mod.rs

```diff
@@ -478,10 +478,10 @@ pub struct InternalSignal {
 }
 
 impl InternalSignal {
-    pub fn new(annotation: String) -> InternalSignal {
+    pub fn new<S: Into<String>>(annotation: S) -> InternalSignal {
         InternalSignal {
             id: uuid(),
-            annotation: Box::leak(annotation.into_boxed_str()),
+            annotation: Box::leak(annotation.into().into_boxed_str()),
```

Comment:
Ok, I should be more consistent with this.

Reply:
I cannot do this now because the InterNal signal needs to implement the Copy trait, but I will be able to do this in the future. Having a Symbol Table that references the signal instead of the other way around.